nixos/modules/system/boot/kexec.nix

```diff
@@ -13,6 +13,9 @@
         path = [ pkgs.kexectools ];
         script =
           ''
+            # Don't load the current system profile if we already have a kernel loaded
+            [[ 1 = "$(</sys/kernel/kexec_loaded)" ]] && exit
```

Comment:
Not sure if that's important here, but the below uses `sh` style `if`s using `[ ]` instead of `bash` `[[ ]]`; you may want to use the same.

Reply:
There are two parts to this:

- https://mywiki.wooledge.org/BashFAQ/031 I generally prefer `[[` except in contexts where the shell might not be bash (in this case it is!), because it's easier to use correctly
- When it's a single command that's being run conditionally and it behaves correctly with `set -e`, I prefer `&&` (or `||` as the case may be) over `if` because (IMHO) `if` results in unhelpful visual noise.

Reply:
I have no opinion about what's better, my comment was only for the case that `script` does not imply bash.